### PR TITLE
fix(Dockerfile): jekyll install

### PR DESCRIPTION
**Pull request by tevariou**
## Fixes

Fixes https://github.com/arkhn/deployment/issues/532 by @nriss 

## Description

Change base image in Dockerfile to fix build error

PR opened to fix os type check fail upstream https://github.com/FHIR/sample-ig/pull/54

## Definition of Done

- [x] I followed the [Arkhn Code Book](https://www.notion.so/arkhn/Arkhn-Code-Book-23ac18a8af7343d0a3f61f1c9ef7400c) (I swear!).
- [ ] I have written tests for the code I added or updated.
- [ ] I have updated the documentation according to my changes.
- [ ] I have updated the deployment configuration if needed.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,12 @@
-FROM jruby:9.2.18.0-jdk11 as build-image
+FROM jekyll/jekyll:4.2.0 as build-image
```
Comment (simonvadee) [inline Dockerfile:1]: much better thanks

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
-    && apt-get autoremove --purge -y \
-    && apt-get clean -y \
-    && rm -rf /var/lib/apt/lists/*
-
-
-RUN gem install jekyll
+RUN apk update \
+    && apk upgrade \
+    && apk add --no-cache \
+        build-base
```
Comment (simonvadee) [inline Dockerfile:6]: forgot `\`

**Comment by simonvadee** [comment]
the build is green but it actually failed lol:
> #23 [build-image 12/12] RUN ./_genonce.sh -y
#23 0.084 ./_genonce.sh: line 5: curl: command not found
#23 0.084 Checking internet connection...
#23 0.084 Offline
#23 0.084 -tx n/a
#23 0.084 IG Publisher NOT FOUND in input-cache or parent folder. Please run _updatePublisher. Aborting...
#23 DONE 0.1s

File: Dockerfile [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
 
 RUN gem install jekyll
-
 RUN npm install -g fsh-sushi
 
-WORKDIR /build
-
 COPY _updatePublisher_curl.sh _updatePublisher_curl.sh
 COPY _genonce.sh _genonce.sh
+RUN ./_updatePublisher_curl.sh -y
```
Comment (simonvadee) [inline Dockerfile:22]: any reason why this should be before the COPY statements ?

File: Dockerfile [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
+
 COPY input input
 COPY ig.ini ig.ini
 COPY sushi-config.yaml sushi-config.yaml
 COPY package-list.json package-list.json
 
-RUN ./_updatePublisher_curl.sh -y || echo "ok"
-
+RUN chmod +x _genonce.sh
```
Comment (simonvadee) [inline Dockerfile:29]: whyn doesn't this file already have correct permissions ?

File: _updatePublisher_curl.sh [hunk 5]
```diff
@@ -32,7 +32,7 @@ done
 
 echo "Checking internet connection"
 case "$OSTYPE" in
-  linux-gnu* ) curl -sSf tx.fhir.org > /dev/null ;;
+  linux* ) curl -sSf tx.fhir.org > /dev/null ;;
```
Comment (simonvadee) [inline _updatePublisher_curl.sh:35]: do we still need both updatePublisher and updatePublisher_curl ?

Comment (tevariou) [reply _updatePublisher_curl.sh:35]: I still have the issue with ping :/

**Review by simonvadee — APPROVED** [review]
lgtm, sorry for the delay!

**Pull request merged**